libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -1027,3 +1060,347 @@ void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
      * from the server to complete the incoming PUBLISH QoS2 transaction. */
     TEST_ASSERT_EQUAL( MQTT_PACKET_ID_INVALID, context.incomingPublishRecords[ 0 ].packetId );
 }
+
+/**
+ * @brief Verifies the behavior of the MQTT library in a restored session with the broker
+ * for resending QoS 1 PUBLISH packets that were un-acknowledged in the previous connection.
+ * Tests that the library is able to support resending the PUBLISH packet with the DUP flag
+ * set in the restored session.
+ */
+void test_MQTT_Restore_Session_Resend_Unacked_Publish_QoS1( void )
+{
+    /* Terminate TLS session and TCP connection network connection to discard current MQTT session
+     * that was created as a "clean session". */
+    ( void ) Openssl_Disconnect( &networkContext );
+
+    /* Establish a new MQTT connection over TLS with the broker with the "clean session" flag set to 0
+     * to start a persistent session with the broker. */
+
+    /* Create the TLS+TCP connection with the broker. */
+    TEST_ASSERT_EQUAL( OPENSSL_SUCCESS, Openssl_Connect( &networkContext,
+                                                         &serverInfo,
+                                                         &opensslCredentials,
+                                                         TRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                                         TRANSPORT_SEND_RECV_TIMEOUT_MS ) );
+    TEST_ASSERT_NOT_EQUAL( -1, networkContext.socketDescriptor );
+    TEST_ASSERT_NOT_NULL( networkContext.pSsl );
+
+    /* Establish a new MQTT connection for starting a persistent session with the broker
+     * by setting the "clean session" flag to 0. */
+    establishMqttSession( &context, &networkContext, false, &persistentSession );
+    TEST_ASSERT_FALSE( persistentSession );
```

Comment:
I don't think this part is necessary. Resending a publish is from the client side, and I'd expect it to be the same whether or not the session actually was resumed by the broker. Also, since this is QoS 1, the broker is not going to have any records of this publish to begin with, regardless if it's a clean or dirty session.

Reply:
Yeah, it's not necessary for the duplicate packet to be sent only for unacked PUBLISH packet in a persistent session. Though, I am not sure how the state transition would behave when a PUBLISH packet is resent even though the first attempt had succeeded.